kentikapi/internal/api_payloads/custom_application.go

```diff
@@ -41,9 +42,9 @@ type CustomApplicationPayload struct {
 	ASN         *string `json:"asn,omitempty"`
 
 	// following fields can appear in request: none, response: get/post/put
-	ID          *models.ID `json:"id,omitempty" response:"get,post,put"`
-	CompanyID   *models.ID `json:"company_id,string,omitempty" response:"get,post,put"`
-	UserID      *models.ID `json:"user_id,string,omitempty"`           // user_id happens to be returned as null
+	ID          *int       `json:"id,omitempty" response:"get,post,put"`
```

Comment:
I don't see a point of having a pointer type for read-only fields such as ID and CompanyID. They are not exposed to the user as optional values. CustomApplicationToPayload() omits those fields, so having "omitempty" ensures they are not encoded to the payload.

> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string. 

I don't see a value in response validation and it seems to be a purpose of that. Removing pointer would disable that validation
@mateuszmidor Could you help me with my confusion? Is that for validation and is it worth it?

Reply:
Pointer type carries the explicit information that this field is optional in the API and this approach is consistent with other fields and other resource types in the library.
Validation carries information how the API is expected to behave.
I think it is better this way

Reply:
> Pointer type carries the explicit information that this field is optional in the API
Exactly. But this field is not optional. It is required to be returned by the API. So the pointer type is misleading here.

Reply:
It is required in Response, but not used in Request. Probably request and response payloads should be separate objects...

Reply:
I don't see a problem that separation of payload types would resolve.

Reply:
The nature of data should be reflected in the respective data types, not in comments; doing otherwise has already proved problematic in python sdk public interface.
Anyway, I would keep fields that are sometimes optional and sometimes required as pointers. Empty string != nil, even though json marshaller can treat them as equal.

Reply:
> The nature of data should be reflected in the respective data types, not in comments;

It is reflected in data types + functions. CustomApplicationToPayload() omits read-only fields so that they are not sent to the server. Comment is just additional info for devs to explain why this function omits them.

I think I understand your point of view now. You're treating this field as optional for requests.
I don't. I treat this field as omitted in requests. And omit-empty mechanism with appropriate CustomApplicationToPayload()  function is enough for me, because it just works.

Anyways, I think I have better understanding of the matter now. I don't plan to modify this code in near future, so we can hold that discussion.

Reply:
Having separate stucts for requests and responses looks reasonable to me.